docs/develop/dialog-api-in-office-add-ins.md

```diff
@@ -207,47 +208,98 @@ function processMessage(arg) {
 
 ## Pass information to the dialog box
 
-Sometimes the host page needs to pass information to the dialog box. You can do this in two primary ways:
+Your add-in can send messages from the [host page](dialog-api-in-office-add-ins.md#open-a-dialog-box-from-a-host-page) to a dialog box using the [messageChild](/javascript/api/office/office.dialog#messagechild-message-) method of the [Dialog](/javascript/api/office/office.dialog) object.
 
-- Add query parameters to the URL that is passed to `displayDialogAsync`.
-- Store the information somewhere that is accessible to both the host window and dialog box. The two windows do not share a common session storage, but *if they have the same domain* (including port number, if any), they share a common [Local Storage](https://www.w3schools.com/html/html5_webstorage.asp).\*
+> [!Note]
+> The APIs are supported in only Excel, PowerPoint, and Word. Support for Outlook is coming soon.
 
-> [!NOTE]
-> \* There is a bug that will effect your strategy for token handling. If the add-in is running in **Office on the web** in either the Safari or Edge browser, the dialog box and task pane do not share the same Local Storage, so it cannot be used to communicate between them.
+### Use `messageChild()` from the host page
+
+When you call the Office dialog API to open a dialog box, a [Dialog](/javascript/api/office/office.dialog) object is returned. It should be assigned to a variable, which typically has greater scope than the [displayDialogAsync](/javascript/api/office/office.ui#displaydialogasync-startaddress--callback-)
+method because the object will be referenced by other methods. The following is an example:
 
-### Use local storage
+```javascript
+var dialog;
+Office.context.ui.displayDialogAsync('https://myDomain/myDialog.html',
+    function (asyncResult) {
+        dialog = asyncResult.value;
+        dialog.addEventHandler(Office.EventType.DialogMessageReceived, processMessage);
+    }
+);
 
-To use local storage, your code calls the `setItem` method of the `window.localStorage` object in the host page before the `displayDialogAsync` call, as in the following example:
+function processMessage(arg) {
+    dialog.close();
 
-```js
-localStorage.setItem("clientID", "15963ac5-314f-4d9b-b5a1-ccb2f1aea248");
+  // message processing code goes here;
+
+}
 ```
 
-Code in the dialog box reads the item when it's needed, as in the following example:
+This `Dialog` object has a [messageChild](/javascript/api/office/office.dialog#messagechild-message-) method that sends any string, or stringified data, to the dialog box. This raises a `DialogParentMessageReceived` event in the dialog box. Your code should handle this event, as shown in the next section.
```

Comment:
I think we're inconsistent about this, but sometimes we surround enum values with quotation marks ("DialogParentMessageReceived"). I guess that makes more sense in Excel because the Rich API overloads methods that use enums with string literal variants.

Anyway, I'm sure if you have to change this. Just something to be aware of.